src/Config.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\cloudflare\Config.
+ */
+
+namespace Drupal\cloudflare;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Psr\Log\LoggerInterface;
+
+use CloudFlarePhpSdk\Exceptions\CloudFlareHttpException;
+use CloudFlarePhpSdk\Exceptions\CloudFlareApiException;
+use CloudFlarePhpSdk\ApiEndpoints\ZoneApi;
+
+/**
+ * Invalidation methods for CloudFlare.
+ */
+class Config implements CloudFlareConfigInterface {
+  /*
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $config;
+
+  /*
+   * @var string
+   */
+  protected $apiKey;
+
+  /*
+   * @var string
+   */
+  protected $email;
+
+  /*
+   * @var string
+   */
+  protected $zone;
+
+  /*
+   * @var \CloudFlarePhpSdk\ApiEndpoints\ZoneApi
+   */
+  protected $zoneApi;
+
+  /*
+   * @var \Psr\Log\LoggerInterface
+   */
+  protected $logger;
+
+  /**
+   * CloudFlareInvalidator constructor.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config
+   *   The config factory.
+   * @param \Psr\Log\LoggerInterface $logger
+   *   A logger instance.
+   */
+  public function __construct(ConfigFactoryInterface $config, LoggerInterface $logger) {
+    $this->config = $config->get('cloudflare.settings');
+    $this->logger = $logger;
+
+    $this->apiKey = $this->config->get('apikey');
+    $this->email = $this->config->get('email');
+    $this->zoneApi = new ZoneApi($this->apiKey, $this->email);
+    if ($this->hasValidApiCredentials()) {
+      $this->zone = $this->getCurrentZoneId();
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getLogger() {
+    return $this->logger;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getZoneApi() {
+    return $this->zoneApi;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCurrentZoneId() {
+    $has_zone_in_cmi = !is_null($this->config->get('zone'));
+
+    // If this is a multi-zone cloudflare account and a zone has been set.
+    if ($has_zone_in_cmi) {
+      return $this->config->get('zone');
+    }
+
+    if (!is_null($this->zone)) {
+      return $this->zone;
+    }
+
+    // If there is no zone set and the account only has a single zone.
+    try {
+      $zones_from_api = $this->zoneApi->listZones();
+    }
+
+    catch (CloudFlareHttpException $e) {
+      drupal_set_message("CloudFlare: Unable to list zones." . $e->getMessage(), 'error');
```

Comment:
It seems these are supposed to be pure API classes. `drupal_set_message()` is a UI thing.

Therefore this doesn't belong here, but in the API code calling this.